src/envs/behavior.py

```diff
@@ -329,7 +429,58 @@ def _bddl_predicate_arity(bddl_predicate: "bddl.AtomicFormula") -> int:
         raise ValueError("BDDL predicate has unexpected arity.")
 
     @staticmethod
-    def _create_type_combo_name(original_name: str,
-                                type_combo: Sequence[Type]) -> str:
+    def _create_type_combo_name(original_name: str,\
+        type_combo: Sequence[Type]) -> str:
         type_names = "-".join(t.name for t in type_combo)
         return f"{original_name}-{type_names}"
+
+
+# NOTE: This function must be type ignored because the env argument is untyped.
```

Comment:
I think it should be possible to work around this by putting the types in quotes. See `object_to_ig_object(self, obj: Object) -> "ArticulatedObject":`, for example